### ISSUE TITLE
Incorrect handling of parameter names that are also macros in trait member generation

### ISSUE THREAD

**Issue by fennewald** [rust-lang/rust-analyzer#21070]
Using `rust-analyzer 1.92.0-nightly (be0ade2 2025-10-11)`, and the following snippet:

```rust
trait Foo {
    fn foo(&self, vec: usize);
}

struct Bar;

impl Foo for Bar {}
```

Select `Foo` in the `impl Foo for Bar`, and choose "Implement missing members". `rust-analyzer` will produce this snippet:
```rust
impl Foo for Bar {
    fn foo(&self, std::vec: usize) {
        todo!()
    }
}
```
The argument named `vec` for foo was incorrectly inserted as `std::vec`.

Notably, this behavior does not occur for other, similar `std::...` names. I tested:
* `alloc`
* `any`
* `arch`
* `array`
* `box`

**Comment by fennewald** [comment]
Wait a second, I think I've figured it out.

The error also occurs for `format` (it is inserted as `std::format`), `panic` and `assert_eq` -- it looks like all parameter names that happen to be global macros exhibits the behavior.

Notably, creating a module and defined a macro manually does _not_ trigger the behavior.

**Comment by fennewald** [comment]
I've tried to reproduce this in an internal test, but it isn't triggering :/

I tried adding a `vec` minicore flag with appropiate macros, but I still couldn't trigger the issue. Maybe I was unsuccessful because minicore doesn't use the `std` prefix the same?

**Comment by ChayimFriedman2** [comment]
If I had to guess, this is related to `#[macro_export]` macros and perhaps the prelude imports. So you need to add a `#[macro_export]` macro to minicore.
